src/args.rs

```diff
@@ -117,6 +117,10 @@ struct CliArgs {
     /// Shown instead of host in page title and heading
     #[structopt(short = "t", long = "title")]
     title: Option<String>,
+
+    /// Custom header from user
+    #[structopt(long = "header")]
+    header: Option<String>,
```

Comment:
I think it would be cool if you'd move the header parsing here instead. This way, a user will receive much better error feedback in case of a bad format. In order to accomplish this, you can change the type to `Option<Header>` here and implement the parsing logic in a [parse](https://docs.rs/structopt/0.3.21/structopt/#custom-string-parsers) function.

Reply:
The logic of parsing header has been moved to args.rs as you suggested.

I reinvent the owned version of Header from [httparse](https://github.com/seanmonstar/httparse) or I need to add lifetime to all config struct. Not sure whether it is a good decision or not.

Thanks for reviewing the code.

Reply:
Oh, sorry didn't see this comment before asking why we need a custom `Header`. Let me think about this a little. Certainly adding a lifetime to our config struct would make it really annoying to use!